designs/inline-io.md

```diff
@@ -0,0 +1,228 @@
+# Inline Operation Inputs / Outputs
+
+This document describes a way to write input and output shapes as an inline
+part of an operation’s definition.
+
+## Motivation
+
+Operation input and output shapes are always structures, almost always have
+boilerplate names, and critically are almost never re-used. In some cases, they
+may not even have types generated for them. Because of those properties, the
+need to fully define them separately from an operation can feel like needless
+boilerplate. Additionally, separating them makes reading an operation at a high
+level more difficult since you have to jump around to get the information you
+need.
+
+## Proposal
+
+Operations will allow inlining input and output definitions, indicated by a
+walrus operator (`:=`). Inlined structures will deviate from normal
+structure definitions in two respects. Firstly, the structure keyword will be
+omitted. Additionally, providing a name for the inlined shape will be optional.
```

Comment:
>Additionally, providing a name for the inlined shape will be optional.

Is this still true? You can only influence the name, not provide one.

Reply:
Yes, this is still true. You can still fully override the name:

```
operation Operation {
    input := Foo {}
}
```